src/main/java/edu/kis/powp/jobs2d/command/visitor/TransformationFlipVisitorImpl.java

```diff
@@ -0,0 +1,51 @@
+package edu.kis.powp.jobs2d.command.visitor;
+
+import edu.kis.powp.jobs2d.command.ICompoundCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+
+public class TransformationFlipVisitorImpl implements TransformationFlipVisitor {
+	private boolean flipX;
+	private boolean flipY;
+	private int x;
+	private int y;
+
+	public TransformationFlipVisitorImpl(boolean flipX, boolean flipY) {
+		this.flipX = flipX;
+		this.flipY = flipY;
+	}
+
+	@Override
+	public void visit(OperateToCommand operateToCommand) {
+		this.x = operateToCommand.getPosX();
+		this.y = operateToCommand.getPosY();
+		this.flip(this.flipX, this.flipY);
+	}
+
+	@Override
+	public void visit(SetPositionCommand setPositionCommand) {
+		this.x = setPositionCommand.getPosX();
+		this.y = setPositionCommand.getPosY();
+		this.flip(this.flipX, this.flipY);
+	}
```

Comment:
implement visitor in the methods below. doForCompundCommand needs to be implemented!